aggregators/internal/hdrhistogram/hdrhistogram.go

```diff
@@ -120,24 +122,21 @@ func (h *HistogramRepresentation) Buckets() (int64, []int64, []float64) {
 	values := make([]float64, 0, h.CountsRep.Len())
 
 	var totalCount int64
-	var bucketsSeen int
+	var prevBucket int32
 	iter := h.iterator()
-	// TODO @lahsivjar: Assuming sparse representation, we might be better off
-	// by sorting CountsRep rather than what we do now to avoid sorting.
-	for idx := 0; iter.next(); idx++ {
-		if bucketsSeen == h.CountsRep.Len() {
-			break
+	iter.nextCountAtIdx()
```

Comment:
I'm wondering if the `iterator` type is still needed, and whether we could simplify things. Would it be possible to store the bucket value, rather than index, in CountsRep? Then `Buckets` is simply iterating through and copying to slices.

Reply:
> Would it be possible to store the bucket value, rather than index, in CountsRep?

Do you mean save the `highestEquivalentValue` for each bucket instead of index? That value is currently calculated based on the bucket and sub bucket index and I think it would require some re-implementation of the core logic.

Reply:
I might be able to use the `bucketIdx` to derive the `highestEquivalentValue` directly without requiring the `subBucket` (which is the whole reason I kept the iterator). Giving this a shot now.

Reply:
:+1: let's take it to a followup